modules/imgproc/src/accum.cpp

```diff
@@ -1493,10 +1182,24 @@ struct AccW_SIMD<ushort, double>
 #endif //CV_SIMD128_64F
 #endif //CV_SIMD128
 
+#if CV_AVX
+template <typename T, typename AT> int Acc_AVX(const T *, AT *, const uchar *, int, int);
+template <typename T, typename AT> int AccSqr_AVX(const T *, AT *, const uchar *, int, int);
+template <typename T, typename AT> int AccProd_AVX(const T *, const T *, AT *, const uchar *, int, int);
+template <typename T, typename AT> int AccW_AVX(const T *, AT *, const uchar *, int, int, AT);
+#endif
+
 template<typename T, typename AT> void
 acc_( const T* src, AT* dst, const uchar* mask, int len, int cn )
 {
-    int i = Acc_SIMD<T, AT>()(src, dst, mask, len, cn);
+    int i;
+    
+#if CV_AVX
+    if (checkHardwareSupport(CPU_AVX))
+        i = Acc_AVX<T, AT>(src, dst, mask, len, cn);
```

Comment:
Please rebase and make changes similar to this patch:
https://github.com/opencv/opencv/pull/9007/files

Currently "dispatching" doesn't work at all.